fuzzyfinder/main.py

```diff
@@ -28,4 +33,12 @@ def fuzzyfinder(input, collection, accessor=lambda x: x):
             best = min(r, key=lambda x: len(x.group(1)))   # find shortest match
             suggestions.append((len(best.group(1)), best.start(), accessor(item), item))
 
-    return (z[-1] for z in sorted(suggestions))
+    if alpha_num_sort:
+        return (z[-1] for z in sorted(suggestions))
+    else:
+        if isinstance(collection[0], str):
+            # Items of `collection` are strings; they would be used for
+            # alphanumeric sorting. Hence remove them when sorting
+            return (z[-1] for z in sorted(suggestions, key=lambda x: x[:2]))
+        else:
+            return (z[-1] for z in sorted(suggestions, key=lambda x: (x[0], x[1], x[3])))
```

Comment:
This whole `else` section should be 
```
return (z[-1] for z in sorted(suggestions, key=lambda x: x[:2]))
```

Because regardless of whether the collection has strings or not we shouldn't be using them for sorting if alpha_num_sort is False.

Reply:
What if the objects of `collection` have custom comparison operators defined ? Then using them for comparison would not be alphanumeric sorting. It's kind of an edge case, but it *is* a scenario.

Reply:
That's an excellent point. My suggestion is to call that parameter `sort_results` instead of calling it `alpha_num_sort`. 

The spirit of this change is to enable/disable sorting regardless of whether it is a string or an object.

Reply:
OK, but calling the parameter `sort_results` might be a bit misleading - the user might think that it controls sorting based on shortest match and position.

Reply:
The problem I have with the current implementation is "surprising behavior". It should have consistent behavior and changing that behavior based on the type of the data seems confusing.

Reply:
I know it's stupid to ask this after this much, but are we disabling sorting as a whole, or just the alpha-numeric sorting, where sorting based on position, etc, is still intact ?

Reply:
No. I'm glad you're sticking to your guns. 

We will not be disabling the rest of the sorting based on rank. We will only be disabling the alpha-numeric sorting. 

Let's go back to the basics and talk about the reasoning. We apply the regex and filter the list, but the list is not ordered optimally. In order to create an optimal ordering, we choose the match length and the starting position. If there are two entries that have the exact same match length and starting position, then we need a tie-breaker. That's where the final sort comes in to decide which one will go first. 

The `sort_results` param or whatever we choose to call this argument is supposed to control the final tie-breaker because that causes problems in certain situations. That's the way I see it. 

That's why I'm recommending that the argument should apply the logic to the entire last comparison and shouldn't discriminate between alphanumerics and other objects.

Reply:
Thanks for clarifying :sweat_smile: